com.microsoft.mrtk.uxcomponents/Tests/Runtime/CanvasDialogTests.cs

```diff
@@ -262,5 +263,100 @@ public IEnumerator TestPoolingRudely()
             Assert.AreEqual(5, dialogs.Length, "There should have been 5 total dialogs used.");
             
         }
+
+        /// <summary>
+        /// Tests using async methods for awaiting on dialog results.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator TestAsyncDialog()
+        {
+            Task<bool> task = AsyncTestDialog();
+
+            // Ew!
+            while (!task.IsCompleted)
+            {
+                yield return null;
+            }
+
+            Assert.IsTrue(task.Result, "Something in the dialog test failed.");
+        }
+
+        private async Task<bool> AsyncTestDialog()
+        {
+            bool wasDismissed = false;
+
+            var dialog = spawner.Get()
+                .SetHeader("This is a test header.")
+                .SetBody("This is a test body.")
+                .SetNeutral("OK");
+
+            dialog.OnDismissed += ( args ) => { wasDismissed = true; };
+
+            // Fire off an async click. This is pretty gross, don't do this outside of testing.
+            var _ = ClickAsync(dialog as Dialog);
```

Comment:
Can we just do `ClickAsync(dialog as Dialog);`?

Reply:
No, compiler complains 🙃 Yes, it's dumb, hence, "gross"